### ISSUE TITLE
Error com.adobe.acs.commons.redirects.filter.RedirectFilter() : Field urlAdjuster in class class com.adobe.acs.commons.redirects.filter.RedirectFilter must be declared volatile to handle a dynamic reference

### ISSUE THREAD

**Issue by kwin** [Adobe-Consulting-Services/acs-aem-commons#2548]
## Required Information

* [x] AEM Version, including Service Packs, Cumulative Fix Packs, etc: AEM Cloud Quickstart 2021.2.4944.20210221T230729Z-210128
* [x] ACS AEM Commons Version: 5.0.2
* [x] Reproducible on Latest? yes

### Expected Behavior

There is no error logged during startup.

### Actual Behavior

The following error is logged
```
[ERROR] 14.03.2021 15:45:51.880 *ERROR* [FelixLogListener] com.adobe.acs.commons.redirects.filter.RedirectFilter bundle com.adobe.acs.acs-aem-commons-bundle:5.0.2 (555)[com.adobe.acs.commons.redirects.filter.RedirectFilter(3181)] : Field urlAdjuster in class class com.adobe.acs.commons.redirects.filter.RedirectFilter must be declared volatile to handle a dynamic reference
[ERROR] 14.03.2021 15:45:51.935 *ERROR* [FelixStartLevel] com.adobe.acs.commons.redirects.filter.RedirectFilter Failed to get resolver for redirect-manager
```

### Steps to Reproduce

Just install ACS AEM Commons 5.0.2 and watch the log while restarting AEM.

**Comment by kwin** [comment]
@YegorKozlov Can you have a look?

**Comment by YegorKozlov** [comment]
Sure @kwin . I'll take a look.

**Comment by kwin** [comment]
Would also be good to use an opt-in approach for the filter, i.e. require an OSGi configuration before it gets active (compare with https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/CONTRIBUTING.md#before-contributing-code):
> Until explicitly enabled, features should be invisible to AEM users and excluded from any execution stack.
OSGi Services automatically registered as part of the stack should be annotated with policy = ConfigurationPolicy.REQUIRE. Common candidates include (but not limited to): Filters, Scheduled Services, Event Listeners and Authentication handlers.
Ex. Until a OSGi Configuration is added for the feature "Component Error Handler", this Sling Filter will remain inactive making it impossible to effect Request processing.

**Comment by abhinavsinha347** [comment]
@kwin @YegorKozlov can this be fixed on VanityUrlServiceImpl as well for Vanity Url adjuster ?

**Comment by davidjgonzalez** [comment]
Available in 5.0.4 (released today)

**Comment by YegorKozlov** [comment]
this can be resolved.  The urlAdjuster  and vanityUrlAdjuster fixes were included in v5.0.4
